w3af/core/ui/api/resources/sessions.py

```diff
@@ -0,0 +1,343 @@
+                plugin_opts[opt_name].set_value([URL(o) for o in opt_value])
+            else:
+                plugin_opts[opt_name].set_value(opt_value)
+        except (AttributeError, BaseFrameworkException):
+            return jsonify({
+                'code': '422',
+                'message': 'Invalid %s option %s' % (opt_type, opt_value)
+                }), 422
+
+    with SET_OPTIONS_LOCK:
+        if 'enabled' in request.json:
+            enable_or_disable_plugin(w3af,
+                                     plugin,
+                                     plugin_type,
+                                     enable=request.json['enabled'])
+        w3af.plugins.set_plugin_options(plugin_type, plugin, plugin_opts)
+
+        return jsonify({
+            'message': 'success',
+            'modified': request.json
+            })
+
+
+@app.route('/sessions/<int:scan_id>/core/<string:core_setting>/',
+           methods=['GET'])
+@requires_auth
+@check_session_exists
+def get_core_settings(scan_id, core_setting):
+    """
+    Shows available core settings and their current values.
+
+    :return: 404 if scan does not exist.
+             Otherwise, a JSON object containing settings and basic help text.
+    """
+    w3af = SCANS[scan_id].w3af_core
+
+    core_setting = core_setting.lower()
+    if core_setting == 'http':
+        opts = w3af.uri_opener.settings.get_options()
+    elif core_setting == 'misc':
+        opts = MiscSettings().get_options()
+    elif core_setting == 'target':
+        opts = w3af.target.get_options()
+    else:
+        return jsonify({
+            'code': 404,
+            'message': '%s is not a valid core setting type' % core_setting }), 404
+
+    opt_dict = { i.get_name():
+            { "value": str(i.get_value()),
+              "description": i.get_desc(),
+              "type": i.get_type(),
+              "default": str(i.get_default_value()) }
+        for i in opts }
+
+    return jsonify({ '%s settings' % core_setting : opt_dict })
+
+
+@app.route('/sessions/<int:scan_id>/core/<string:core_setting>/',
+           methods=['PATCH'])
+@requires_auth
+@check_session_exists
+def set_core_config(scan_id, core_setting):
+    """
+    Allows a user to modify core configuration by sending a PATCH request.
+
+    :return: 404 if scan does not exist.
+             400 if the PATCH request is not in valid format.
+             422 if the PATCH request is in valid format, but settings cannot be
+             applied (eg, negative numbers where none are accepted).
+
+             Otherwise, a JSON object confirming success and echoing the
+             changed setting.
+    """
+    w3af = SCANS[scan_id].w3af_core
+
+    if core_setting == 'http':
+        configurable = w3af.uri_opener.settings
+    if core_setting == 'misc':
+        configurable = MiscSettings()
+    if core_setting == 'target':
+        configurable = w3af.target
+
+    core_opts = configurable.get_options()
```

Comment:
The next ~30 lines look very much like the configuration of plugin parameters. Maybe you should avoid code duplication and move this to a helper?